### DISCUSSION TITLE
Why is "this" untyped when my store is defined in a TypeScript file?

### DISCUSSION THREAD

**Discussion by leddt** [vuejs/pinia#899] [Help and Questions] [answered]
Take for example this simple store:
```js
import { defineStore } from "pinia";

export const useAppStore = defineStore("app", {
  state: () => ({
    showMenu: true
  }),
  actions: {
    toggleMenu() {
      this.showMenu = !this.showMenu;
    }
  }
});
```

If this code is in a JS file, everything works great: `this` is typed in `toggleMenu`.
![image](https://user-images.githubusercontent.com/239219/145915673-c99c9f7c-3b5b-404c-becd-9e3b44f27b56.png)

If I change the file extension to `.ts` instead, `this` is now typed as `any`.
![image](https://user-images.githubusercontent.com/239219/145915630-6bfbba9a-6d60-4cf6-b907-1097e1d05d90.png)

**Comment by leddt** [comment, marked as answer]
Nevermind, I figured it out. I had `"strict": false` in my tsconfig. Setting it to true fixes it.

**Comment by wildwind123** [comment]
option
`"strict":true` 
enable seven options
`
noImplicitAny
noImplicitThis
alwaysStrict
strictBindCallApply
strictNullChecks
strictFunctionTypes
strictPropertyInitialization
`
for type this.property. 
you can enable one option.
`"noImplicitThis": true`

**Comment by gryphonmyers** [comment]
This does not work when using `"module": "NodeNext"`
